server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2227,6 +2229,7 @@ public void deletePools(Collection<Pool> pools, Collection<String> alreadyDelete
                 }
 
                 this.poolCurator.updateAll(poolsToSave, false, false);
+                this.poolCurator.batchDetach(poolsToSave);
```

Comment:
This one concerns me a bit, as there's one code path where we call into deletePools but it's not the last step of the path. If we detach these pools which are being saved here and they happen to be ones the caller is operating on, and the caller makes changes that are expected to be automatically persisted at the end of the transaction, we are broken.

I'll have to step through that path manually to see if that's a concern, but it's definitely something to note.

Reply:
I will remove this detach.  It's not essential and I didn't see it was passed in as a parameter.